Automattic/vip-go-mu-plugins
# Report correct name during settings check

## Description

The reports we get for search settings inconsistencies would have incorrect index names. That happens if the non-active index has the issue. The reason behind this is that we would reset the current index to active BEFORE we would fetch the index name.

The fix here moves stuff around to make sure we get the name when the test with inconsistencies is still current.

## Changelog Description

### Plugin Updated: Enterprise Search

Internal index settings reporting tweak.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1) Make sure you have 2 versions of index - `dev-env exec -- wp vip-search index-versions add post` and `dev-env exec -- wp vip-search index --setup --version=2 --indexables=post`
2) Make settings inconsistent. For example - change `filter__ep_default_index_number_of_shards` to return different value.
3) Run: `dev-env exec -- wp eval "var_dump((new Automattic\VIP\Search\Health(\Automattic\VIP\Search\Search::instance()))->get_index_settings_health_for_all_indexables());";`
4) Make sure the index names are correctly reported.
